pramen/core/src/main/scala/za/co/absa/pramen/core/reader/TableReaderJdbc.scala

```diff
@@ -109,7 +109,8 @@ class TableReaderJdbc(jdbcReaderConfig: TableReaderJdbcConfig,
 
   private[core] def getCountForSql(sql: String, infoDateBegin: LocalDate, infoDateEnd: LocalDate): Long = {
     val filteredSql = TableReaderJdbcNative.getFilteredSql(sql, infoDateBegin, infoDateEnd)
-    getWithRetry[Long](filteredSql, isDataQuery = false, jdbcRetries, None)(df => df.count())
+    val countSql = s"SELECT COUNT(*) AS CNT FROM ($filteredSql)"
```

Comment:
```suggestion
    val countSql = s"SELECT COUNT(*) FROM ($filteredSql)"
```

Reply:
This won't work because of Spark. We have to explicitly define a column alias, otherwise spark is not able to map the resulted set to dataframe.

Reply:
Yes, checked. It didn't work indeed. Okay, the issue is that not every SQL dialect of sources we work with support aliasing with `AS`.

Here is the new proposal:
```suggestion
    val columnAlias = sqlGen.getAliasExpression("COUNT(*)", "CNT")
    val countSql = s"SELECT $columnAlias FROM ($filteredSql)"
```